### PR TITLE
chore(playgrounds): add all types playground for type testing

### PR DESCRIPTION
**Pull request by Anemy**
Written by @lerouxb 
Figured this would be helpful for testing down the line so it makes sense to be in a code base.

Here's what the resulting document schema looks like in our view:
Screen Shot 2023-01-04 at 11 33 17 AM

### CODE REVIEW

File: playgrounds/all-field-types.mongodb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+// Create a document with all the BSON types.
+const doc = {
+  double: new Double(1.2), // Double, 1, double
+  string: 'Hello, world!', // String, 2, string
+  object: { key: 'value' }, // Object, 3, object
+  array: [1, 2, 3], // Array, 4, array
+  binData: new Binary(Buffer.from([1, 2, 3])), // Binary data, 5, binData
+  // Undefined, 6, undefined (deprecated)
+  objectid: new ObjectId(), // ObjectId, 7, objectId
```
Comment (lerouxb) [inline playgrounds/all-field-types.mongodb:13]: Do you mind changing this to objectId? It is a small thing, but I'm trying to match the aliases as listed in the documentation https://www.mongodb.com/docs/manual/reference/bson-types/

(things aren't consistently named everywhere and I'm trying to standardise on something)

**Review by lerouxb — APPROVED** [review]

File: playgrounds/all-field-types.mongodb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  boolean: true, // Boolean, 8, boolean
+  date: new Date(), // Date, 9, date
+  null: null, // Null, 10, null
+  regex: new BSONRegExp('pattern', 'i'), // Regular Expression, 11, regex
+  dbref: new DBRef('namespace', new ObjectId()), // DBPointer, 12, dbPointer (deprecated)
```
Comment (lerouxb) [inline playgrounds/all-field-types.mongodb:18]: and this to dbPointer :)

File: playgrounds/all-field-types.mongodb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  object: { key: 'value' }, // Object, 3, object
+  array: [1, 2, 3], // Array, 4, array
+  binData: new Binary(Buffer.from([1, 2, 3])), // Binary data, 5, binData
+  // Undefined, 6, undefined (deprecated)
+  objectId: new ObjectId(), // ObjectId, 7, objectId
+  boolean: true, // Boolean, 8, boolean
+  date: new Date(), // Date, 9, date
+  null: null, // Null, 10, null
+  regex: new BSONRegExp('pattern', 'i'), // Regular Expression, 11, regex
+  dbPointer: new DBRef('namespace', new ObjectId()), // DBPointer, 12, dbPointer (deprecated)
```
Comment (addaleax) [inline playgrounds/all-field-types.mongodb:18]: Sooo … `DBPointer` and `DBRef` are different things :) `DBPointer` is an actual bson type, `DBRef` is a convention for regular BSON documents. The driver’s bson package doesn’t actually support `DBPointer`; when it encounters a `DBPointer` during deserialization, it converts it to `DBRef`.

(`DBRef` isn’t deprecated, either.)

Comment (lerouxb) [reply playgrounds/all-field-types.mongodb:18]: OK I'll make a PR to fix it. I want to add the two missing types anyway. Even though things are deprecated we should probably test to see what happens there.

Comment (lerouxb) [reply playgrounds/all-field-types.mongodb:18]: oh it hasn't been merged so I can just edit this one.

**Review by addaleax — APPROVED** [review]

**Pull request merged**